crates/nu-engine/src/glob_from.rs

```diff
@@ -72,20 +73,32 @@ pub fn glob_from(
         if is_symlink {
             (path.parent().map(|parent| parent.to_path_buf()), path)
         } else {
-            let path = if let Ok(p) = canonicalize_with(path.clone(), cwd) {
-                if p.to_string_lossy().contains(GLOB_CHARS) {
+            let path = match canonicalize_with(path.clone(), cwd) {
+                Ok(p) if p.to_string_lossy().contains(GLOB_CHARS) => {
                     // our path might contains GLOB_CHARS too
                     // in such case, we need to escape our path to make
                     // glob work successfully
                     PathBuf::from(nu_glob::Pattern::escape(&p.to_string_lossy()))
-                } else {
-                    p
                 }
-            } else {
-                return Err(ShellError::DirectoryNotFound {
-                    dir: path.to_string_lossy().to_string(),
-                    span: pattern.span,
-                });
+                Ok(p) => p,
+                Err(err) => {
+                    return match err.kind() {
+                        ErrorKind::PermissionDenied => Err(ShellError::GenericError {
```

Comment:
I think this should be a `ShellError::IOError` instead of a `ShellError::GenericError`; `cd` uses `IOError` when you attempt to `cd` into a directory without permission.

Reply:
Since `ShellError::IOError` doesn't contain an easy reliable way to check which io error is used, I don't like using it. And adding `std::io::ErrorKind` to `ShellError::IOError` which would be optimal, kind of explodes the scope here imo.